UnityProject/Assets/Scripts/Items/Science/PneumaticCannon.cs

```diff
@@ -0,0 +1,259 @@
+		SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.Valve, transform.position, sourceObj: interaction.Performer);
+	}
+
+	#endregion
+
+	#region AimApply Interactions
+
+	public bool WillInteract(AimApply interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+		if (isOnCooldown == false && interaction.MouseButtonState == MouseButtonState.PRESS && interaction.Intent == Intent.Harm) return true;
+
+		return false;
+	}
+
+	public void ServerPerformInteraction(AimApply interaction)
+	{
+		bool fireFailed = false;
+		if(canister == null)
+		{
+			Chat.AddWarningMsgFromServer(interaction.Performer, $"You attempt to fire the cannon {gameObject.ExpensiveName()} no canister is loaded!");
+			fireFailed = true;
+		}
+		else if (canister.ServerInternalPressure < CANNON_MIN_FUNCTIONAL_PRESSURE)
+		{
+			Chat.AddWarningMsgFromServer(interaction.Performer, $"You attempt to fire the cannon {gameObject.ExpensiveName()} the gas pressure was too low!");
+			fireFailed = true;
+		}
+		else if (itemSlot.IsEmpty)
+		{
+			Chat.AddWarningMsgFromServer(interaction.Performer, $"You attempt to fire the {gameObject.ExpensiveName()} but no item was loaded!");
+			fireFailed = true;
+		}
+
+		if(fireFailed)
+		{
+			SoundManager.PlayNetworkedAtPos(dryFiringSound, transform.position, sourceObj: interaction.Performer);
+			return;
+		}
+
+		Chat.AddActionMsgToChat(interaction.Performer,
+					$"You fire your {gameObject.ExpensiveName()}",
+					$"{interaction.Performer.ExpensiveName()} fires their {gameObject.ExpensiveName()}");
+
+		lastShootTime = Time.time;
+		float percentDischarge = Mathf.Clamp(((int)pressureSetting * CANNON_MAX_PRESSURE) / canister.GasMixLocal.Pressure, 0, 1);
+		float pressureToDischarge = Math.Min((int)pressureSetting * CANNON_MAX_PRESSURE, canister.GasMixLocal.Pressure);
+
+		float molesToTransfer = percentDischarge * canister.GasMixLocal.Moles;
+		var metaDataLayer = MatrixManager.AtPoint(interaction.Performer.RegisterTile().WorldPositionServer, true).MetaDataLayer;
+		var metaNode = metaDataLayer.Get(interaction.Performer.RegisterTile().LocalPositionServer);
+
+		GasMix mixToEffect = metaNode.GasMixLocal;
+		GasMix.TransferGas(mixToEffect, canister.GasMixLocal, molesToTransfer);
+
+		SoundManager.PlayNetworkedAtPos(firingSound, transform.position, sourceObj: interaction.Performer);
+		if (pressureToDischarge >= (int)PressureSetting.High * CANNON_MAX_PRESSURE) interaction.PerformerPlayerScript.RegisterPlayer.ServerStun(dropItem: false);
+
+		//We assume a barrel area of 0.1 square meters applied for 1/10th of a second.
+		var physics = itemSlot.ItemObject.GetComponent<UniversalObjectPhysics>();
+		if (physics == null) return;
+
+		Inventory.ServerDrop(itemSlot);
+		physics.NewtonianNewtonPush(interaction.TargetVector, pressureToDischarge * 0.05f, 0.5f, 0.5f, interaction.TargetBodyPart, gameObject, 1f);
```

Comment:
Technically you have to accounts for the pressure outside of the container if you want to be fully accurate, 
gameplay impact your Canon will be more powerful in space versus on the station

Reply:
Would at most be a difference of 100kPa on a 500kPa (weakest setting) shot. Could add but seemed unnessacary

Reply:
ok all good